TU20Bot/Configuration/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using EmbedIO;
+using EmbedIO.Routing;
+using EmbedIO.WebApi;
+using Jose;
+using MongoDB.Driver;
+using TU20Bot.Models;
+
+namespace TU20Bot.Configuration.Controllers {
+    public class AuthenticationController : ServerController {
+
+        [Route(HttpVerbs.Post, "/login")]
+        public string Login([QueryField] string username, [QueryField] string password) {
+
+            var user = server.client.database
+                .GetCollection<AccountModel>(AccountModel.collectionName)
+                .Find(x => x.username == username.ToLower());
+
+            if (!user.Any()) {
+                throw HttpException.Unauthorized();
+            }
+
+            if (!user.First().checkPassword(password))
+                throw HttpException.Unauthorized();
+
+            return JWT.Encode(new AuthorizationPayload {
+                fullName = username,
+                validUntil = DateTime.Now.AddHours(1), // Expire in 1 hour
+                permissions = user.First().permissions
+            }, Encoding.UTF8.GetBytes(server.config.jwtSecret), JwsAlgorithm.HS256);
+        }
+
+        [Route(HttpVerbs.Post, "/create")]
+        public async Task<string> Create([QueryField] string username, [QueryField] string password) {
+
+            if (!AuthorizationModule.validatePermissions(new List<string> { "Admin" }, Request.Headers["Authorization"], server.config.jwtSecret, Response)) {
```

Comment:
This line is a bit long...
```suggestion
            if (!AuthorizationModule.validatePermissions(new List<string> { "Admin" },
                Request.Headers["Authorization"], server.config.jwtSecret, Response)) {
```
I would split this line a bit further. It renders off my screen in review.